### PR TITLE
[FLINK-9299] [Documents] ProcessWindowFunction documentation Java examples have error

### PR DESCRIPTION
**Pull request by medcv**
## What is the purpose of the change

ProcessWindowFunction documentation Java examples have error

## Brief change log

ProcessWindowFunction is an abstract class and should extends instead of implements 

## Does this pull request potentially affect one of the following parts:

  - Dependencies (does it add or upgrade a dependency): (no)
  - The public API, i.e., is any changed class annotated with `@Public(Evolving)`: (no)
  - The serializers: (no)
  - The runtime per-record code paths (performance sensitive): (no)
  - Anything that affects deployment or recovery: JobManager (and its components), Checkpointing, Yarn/Mesos, ZooKeeper: (no)
  - The S3 file system connector: (no)

## Documentation

  - Does this pull request introduce a new feature? (no)
  - If yes, how is the feature documented? (not applicable / docs / JavaDocs / not documented)

### CODE REVIEW

File: docs/dev/stream/operators/windows.md [hunk 1]
```diff
@@ -730,9 +730,9 @@ input
 
 /* ... */
 
-public class MyProcessWindowFunction implements ProcessWindowFunction<Tuple<String, Long>, String, String, TimeWindow> {
+public static class MyProcessWindowFunction extends ProcessWindowFunction<Tuple<String, Long>, String, String, TimeWindow> {
```
Comment (StephanEwen) [inline docs/dev/stream/operators/windows.md]: In other code samples, we don't put the `static`, not assuming that this is defined as an inner class

**Comment by StephanEwen** [comment]
Congrats on having PR number 6000!

This overlaps with #6001, which is a mit more comprehensive (but need some improvements).
Would you be up to coordinate to make one joint PR?

File: docs/dev/stream/operators/windows.md (hunk 1 above)
Comment (medcv) [reply docs/dev/stream/operators/windows.md]: make sense! 
I see some `static` in this docs for other examples, we might need to make them consistent

`private static class MyReduceFunction implements ReduceFunction<SensorReading> {`

**Comment by medcv** [comment]
@StephanEwen Thanks
Yes, I made this PR before asking about ticket status. my bad!
I will close it and will work on #6001

**Pull request closed without merging**